processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -194,16 +207,81 @@ protected void scanAndAggregate(
   )
   {
     final Cursor cursor = params.getCursor();
-    if (specializeGeneric1AggPooledTopN && theAggregators.length == 1) {
-      scanAndAggregateGeneric1Agg(params, positions, theAggregators[0], cursor);
-    } else if (specializeGeneric2AggPooledTopN && theAggregators.length == 2) {
+    if (theAggregators.length == 1) {
```

Comment:
Those `BaseQuery.checkInterrupted();` + `return;` are pretty buried and would be easy to miss in future additions. So having a loop and finishing on those two items feels more sustainable.